src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
 package core.basesyntax;
 
+//import java.util.HashMap;
+import java.util.Objects;
+import java.util.Random;
+
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public double EXPANSION_COEFFICIENT = 2;
+    public double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] values;
+    private static final int defaultCapacity = 16;
+    private int size;
+    private Integer threshold;
+
+    public MyHashMap() {
+        values = new Node[defaultCapacity];
+        this.threshold = (int) (defaultCapacity * LOAD_FACTOR);
+        size = 0;
+    }
+
+    private static class Node<K, V> {
```

Comment:
this is an inner class -> we don't need getters and setters as we have access to them